src/Components/WebView/WebView/src/IpcReceiver.cs

```diff
@@ -89,7 +91,10 @@ private Task DispatchBrowserEventAsync(PageContext pageContext, string eventDesc
         {
             var renderer = pageContext.Renderer;
             var jsonSerializerOptions = pageContext.JSRuntime.ReadJsonSerializerOptions();
-            var webEventData = WebEventData.Parse(renderer, jsonSerializerOptions, eventDescriptor, eventArgs);
+            // JsonSerializerOptions are tightly bound to the JsonContext. Cache it on first use using a copy
+            // of the serializer settings.
+            _jsonContext ??= new(new JsonSerializerOptions(jsonSerializerOptions));
```

Comment:
This seems slightly suboptimal in that it's calling `pageContext.JSRuntime.ReadJsonSerializerOptions()` even when a value is already cached for `_jsonContext`.

Could it skip that call in that case? Same for the other places where this pattern is used.

Reply:
e.g., something like this, which it seems you already do in other places:

```cs
// var jsonSerializerOptions = pageContext.JSRuntime.ReadJsonSerializerOptions(); // Can be removed
_jsonContext ??= new(new JsonSerializerOptions(pageContext.JSRuntime.ReadJsonSerializerOptions()));
```